sensors/triggers/pulsar/pulsar.go

```diff
@@ -0,0 +1,200 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package pulsar
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/apache/pulsar-client-go/pulsar"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+
+	"github.com/argoproj/argo-events/common"
+	"github.com/argoproj/argo-events/common/logging"
+	apicommon "github.com/argoproj/argo-events/pkg/apis/common"
+	"github.com/argoproj/argo-events/pkg/apis/sensor/v1alpha1"
+	"github.com/argoproj/argo-events/sensors/triggers"
+)
+
+// PulsarTrigger describes the trigger to place messages on Pulsar topic using a producer
+type PulsarTrigger struct {
+	// Sensor object
+	Sensor *v1alpha1.Sensor
+	// Trigger reference
+	Trigger *v1alpha1.Trigger
+	// Pulsar async producer
+	Producer pulsar.Producer
+	// Logger to log stuff
+	Logger *zap.SugaredLogger
+}
+
+// NewPulsarTrigger returns a new Pulsar trigger context.
+func NewPulsarTrigger(sensor *v1alpha1.Sensor, trigger *v1alpha1.Trigger, pulsarProducers map[string]pulsar.Producer, logger *zap.SugaredLogger) (*PulsarTrigger, error) {
+	pulsarTrigger := trigger.Template.Pulsar
+
+	producer, ok := pulsarProducers[trigger.Template.Name]
+	if !ok {
+		var err error
+		tlsTrustCertsFilePath := ""
+		if pulsarTrigger.TLSTrustCertsSecret != nil {
+			tlsTrustCertsFilePath, err = common.GetSecretVolumePath(pulsarTrigger.TLSTrustCertsSecret)
+			if err != nil {
+				logger.Errorw("failed to get TLSTrustCertsFilePath from the volume", "error", err)
+				return nil, err
+			}
+		}
+		clientOpt := pulsar.ClientOptions{
+			URL:                        pulsarTrigger.URL,
+			TLSTrustCertsFilePath:      tlsTrustCertsFilePath,
+			TLSAllowInsecureConnection: pulsarTrigger.TLSAllowInsecureConnection,
+			TLSValidateHostname:        pulsarTrigger.TLSValidateHostname,
+		}
+
+		if pulsarTrigger.AuthTokenSecret != nil {
+			token, err := common.GetSecretFromVolume(pulsarTrigger.AuthTokenSecret)
+			if err != nil {
+				logger.Errorw("failed to get AuthTokenSecret from the volume", "error", err)
+				return nil, err
+			}
+			clientOpt.Authentication = pulsar.NewAuthenticationToken(token)
+		}
+
+		if pulsarTrigger.TLS != nil {
+			logger.Info("setting tls auth option...")
+			var clientCertPath, clientKeyPath string
+			switch {
+			case pulsarTrigger.TLS.ClientCertSecret != nil && pulsarTrigger.TLS.ClientKeySecret != nil:
+				clientCertPath, err = common.GetSecretVolumePath(pulsarTrigger.TLS.ClientCertSecret)
+				if err != nil {
+					logger.Errorw("failed to get ClientCertPath from the volume", "error", err)
+					return nil, err
+				}
+				clientKeyPath, err = common.GetSecretVolumePath(pulsarTrigger.TLS.ClientKeySecret)
+				if err != nil {
+					logger.Errorw("failed to get ClientKeyPath from the volume", "error", err)
+					return nil, err
+				}
+			case pulsarTrigger.TLS.DeprecatedClientCertPath != "" && pulsarTrigger.TLS.DeprecatedClientKeyPath != "":
```

Comment:
Those deprecated fields were just removed.

Reply:
I have removed that code now.